### ISSUE TITLE
Rook-Ceph CSI plugins have suddenly crashed and all PVC mounts were down till rook-operator was restarted

### ISSUE THREAD

**Issue by alexcpn** [rook/rook#6601]
* Bug Report

Rook-Ceph CSI plugins have suddenly crashed and all PVC mounts were down till rook-operator was restarted
Happened in two testbed, one in CD and one in Production. Not sure exactly what triggered this. There was no node restart in one testbed at least. For the other testbed not sure.

**Deviation from expected behavior:**

Not sure but is this related to owner reference 

- https://github.com/rook/rook/pull/4788
- https://github.com/rook/rook/pull/4820

In Rook 1.4.1 and 1.4.2 the owner reference is added back? is this related to the above bugs ?
```
 kubectl  -n rook-ceph get  configmap rook-ceph-csi-config -o yaml
apiVersion: v1
data:
  csi-cluster-config-json: '[{"clusterID":"rook-ceph","monitors":["10.3.56.7:6789","10.3.53.81:6789","10.3.87.216:6789"]}]'
kind: ConfigMap
metadata:
  creationTimestamp: "2020-11-07T13:04:56Z"
  name: rook-ceph-csi-config
  namespace: rook-ceph
  ownerReferences:
  - apiVersion: apps/v1 
    blockOwnerDeletion: false
    controller: true
    kind: Deployment
    name: rook-ceph-operator
    uid: 225e2a5c-fd70-11ea-a83e-2e09ef58688b
  resourceVersion: "14641352"
  selfLink: /api/v1/namespaces/rook-ceph/configmaps/rook-ceph-csi-config
  uid: d5810ac8-20f9-11eb-81f5-2e09ef58688b
```
In deployment, the API version looks correct

```
 kubectl -n rook-ceph get deployment rook-ceph-operator -o yaml
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"apps/v1","kind":"Deployment","metadata":{"annotations":{},"labels":{"operator":"rook","storage-backend":"ceph"},"name":"rook-
  generation: 1
  labels:
    operator: rook
    storage-backend: ceph
  name: rook-ceph-operator
  namespace: rook-ceph
  resourceVersion: "14641034"
  selfLink: /apis/extensions/v1beta1/namespaces/rook-ceph/deployments/rook-ceph-operator
  uid: 225e2a5c-fd70-11ea-a83e-2e09ef58688b
```
State of pods in one testbed ,where this was not noticed for some days
```
csi-cephfsplugin-bqvtx                                      0/3     ContainerCreating   0          20d
csi-cephfsplugin-djkc2                                      0/3     ContainerCreating   0          20d
csi-cephfsplugin-lctkb                                      0/3     ContainerCreating   0          20d
csi-cephfsplugin-n4tjr                                      0/3     ContainerCreating   0          20d
csi-cephfsplugin-provisioner-cfffcbf5c-hlh9w                0/6     ContainerCreating   0          20d
csi-cephfsplugin-provisioner-cfffcbf5c-xggtc                0/6     ContainerCreating   0          20d
csi-cephfsplugin-spbcn                                      0/3     ContainerCreating   0          20d
csi-cephfsplugin-wk2qh                                      0/3     ContainerCreating   0          20d
csi-rbdplugin-65qxt                                         0/3     ContainerCreating   0          20d
csi-rbdplugin-dd246                                         0/3     ContainerCreating   0          20d
csi-rbdplugin-dwpk6                                         0/3     ContainerCreating   0          20d
csi-rbdplugin-kksvk                                         0/3     ContainerCreating   0          20d
csi-rbdplugin-provisioner-d9fc4d685-9shdl                   0/6     ContainerCreating   0          20d
csi-rbdplugin-provisioner-d9fc4d685-kmx56                   0/6     ContainerCreating   0          20d
csi-rbdplugin-q9vbq                                         0/3     ContainerCreating   0          20d
csi-rbdplugin-wpkbn                                         0/3     ContainerCreating   0          20d
rook-ceph-crashcollector-dc4-p1-r3-srv-2-7c75fd648d-zf76w   1/1     Running             0          25d
rook-ceph-crashcollector-dc4-p1-r3-srv-3-5698c78997-qk7dd   1/1     Running             0          25d
rook-ceph-crashcollector-dc4-p1-r3-srv-4-55b8b57bfd-pzrsq   1/1     Running             0          25d
rook-ceph-crashcollector-dc4-p1-r3-srv-5-787bd4c57c-wx5l7   1/1     Running             0          25d
rook-ceph-crashcollector-dc4-p1-r3-srv-6-c4c85d59b-nglqq    1/1     Running             0          25d
rook-ceph-crashcollector-dc4-p1-r3-srv-7-dcb8c5c67-7vznj    1/1     Running             0          25d
rook-ceph-mgr-a-8577cf684-x8qqw                             1/1     Running             0          25d
rook-ceph-mon-a-644f9d45b6-kfmtn                            1/1     Running             0          25d
rook-ceph-mon-b-7497c8f444-pf65v                            1/1     Running             0          25d
rook-ceph-mon-c-7d84bfc795-zv5w7                            1/1     Running             0          25d
rook-ceph-operator-658d66cbd-wd4j8                          1/1     Running             0          20d
rook-ceph-osd-0-848894486d-djftx                            1/1     Running             0          25d
rook-ceph-osd-1-7488d8f9f9-z5jzr                            1/1     Running             0          25d
rook-ceph-osd-2-6d7649f867-xm7s5                            1/1     Running             0          25d
rook-ceph-osd-3-5db4986b84-k9gjl                            1/1     Running             0          25d
rook-ceph-osd-4-55c4468c49-d4dt9                            1/1     Running             0          25d
rook-ceph-osd-5-cb9d5b4cb-j2pjp                             1/1     Running             0          25d
rook-ceph-osd-prepare-dc4-p1-r3-srv-2-gb7tb                 0/1     Completed           0          25d
rook-ceph-osd-prepare-dc4-p1-r3-srv-3-7jh5j                 0/1     Completed           0          25d
rook-ceph-osd-prepare-dc4-p1-r3-srv-4-x9bhk                 0/1     Completed           0          25d
rook-ceph-osd-prepare-dc4-p1-r3-srv-5-s2k6v                 0/1     Completed           0          25d
rook-ceph-osd-prepare-dc4-p1-r3-srv-6-s2nwk                 0/1     Completed           0          25d
rook-ceph-osd-prepare-dc4-p1-r3-srv-7-mvggf                 0/1     Completed           0          25d
rook-ceph-rgw-rook-ceph-objstore-a-648f8bbdb5-mhds5         1/1     Running             0          25d
rook-ceph-tools-7d45f5b95-5bcnr                             1/1     Running             1          25d
rook-discover-562kq                                         1/1     Running             0          25d
rook-discover-974d5                                         1/1     Running             1          25d
rook-discover-bg5h6                                         1/1     Running             1          25d
rook-discover-ghtl5                                         1/1     Running             0          25d
rook-discover-j47zm                                         1/1     Running             0          25d
rook-discover-pfhlz                                         1/1     Running             1          25d
 ````
**Expected behavior:**

**How to reproduce it (minimal and precise):**

Tried to reproduce by restarting rook-operator multiple times, but could not.

**File(s) to submit**:

Could not collect operator logs when the incident happened.Have logs only after the operator was restarted.

Checked the KubeAPI server logs for Garbage collection related to CSI - could not find anything related to rook or CSI during those times

```
core@da-cicd-enc1-bl2 ~ $ docker logs k8s_kube-apiserver_kube-apiserver-da-cicd-enc1-bl2_kube-system_9462cfa708281a76ed96d7c74ffe97a3_3  2>&1 | grep "generic-garbage-collector"
I1107 03:13:31.587890       1 trace.go:81] Trace[1497714298]: "Get /api/v1/namespaces/kube-system/serviceaccounts/generic-garbage-collector" (started: 2020-11-07 03:13:31.013508893 +0000 UTC m=+3869224.703926870) (total time: 574.360036ms):
```

**Environment**:
* OS (e.g. from /etc/os-release): CoreOS 1967.6.0 (Rhyolite)   
* Kernel (e.g. `uname -a`): 4.14.96-coreos-r1
* Cloud provider or hardware configuration: Custom bare-metal 
* Rook version (use `rook version` inside of a Rook Pod): rook/ceph:v1.4.1
* Storage backend version (e.g. for ceph do `ceph -v`): ceph/ceph:v15.2.4
* Kubernetes version (use `kubectl version`):  GitVersion:"v1.14.5"
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): Custom baremetal
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):

**Comment by travisn** [comment]
This sounds like an issue that was fixed in v1.4.3. Please see the first item in the [v1.4.3 release notes](https://github.com/rook/rook/releases/tag/v1.4.3).

**Comment by alexcpn** [comment]
Thanks for the quick response. So I guess the owner reference is removed https://github.com/openshift/rook/pull/136 as part of the correction.
I tried rebooting the workers multiple times but could not create the problem. Anyway will update to latest rook
